codec-http/src/main/java/io/netty/handler/codec/http/ServerCookieDecoder.java

```diff
@@ -157,20 +119,48 @@
                     if (i == headerLen) {
                         // NAME (no value till the end of string)
                         newNameEnd = headerLen;
-                        value = null;
+                        newValueStart = newValueEnd = -1;
                         break;
                     }
                 }
             }
 
-            if (!rfc2965Style || (!header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) &&
-                    !header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) &&
-                    !header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
+            if (newValueStart == -1) {
+                LOGGER.debug("Skipping cookie with null value");
+                continue;
+            }
+
+            if (rfc2965Style && (header.regionMatches(newNameStart, "$Path", 0, "$Path".length()) ||
+                    header.regionMatches(newNameStart, "$Domain", 0, "$Domain".length()) ||
+                    header.regionMatches(newNameStart, "$Port", 0, "$Port".length()))) {
 
                 // skip obsolete RFC2965 fields
-                String name = header.substring(newNameStart, newNameEnd);
-                cookies.add(new DefaultCookie(name, value));
+                continue;
             }
+
+            if (containsInvalidCookieNameOctet(header, newNameStart, newNameEnd)) {
+                LOGGER.debug("Skipping cookie because name contains an invalid char");
```

Comment:
This will lead to two log messages because `containsInvalidOctet()` also does logging.  Could we make logging occurs only once?